kubetest/e2e.go

```diff
@@ -626,7 +630,47 @@ func kubemarkTest(testArgs []string, dump, numNodes string) error {
 	// Run tests on the kubemark cluster.
 	if err := control.XmlWrap(&suite, "Kubemark Test", func() error {
 		testArgs = util.SetFieldDefault(testArgs, "--ginkgo.focus", "starting\\s30\\pods")
-		return control.FinishRunning(exec.Command("./test/kubemark/run-e2e-tests.sh", testArgs...))
+
+		// detect master IP
+		if err := os.Setenv("MASTER_NAME", os.Getenv("INSTANCE_PREFIX")+"-kubemark-master"); err != nil {
+			return err
+		}
+
+		masterIP, err := control.Output(exec.Command(
+			"gcloud", "compute", "addresses", "describe",
+			os.Getenv("MASTER_NAME")+"-ip",
+			"--project="+o.gcpProject,
+			"--region="+o.gcpZone[:len(o.gcpZone)-2],
+			"--format=value(address)"))
+		if err != nil {
+			return fmt.Errorf("failed to get masterIP: %v", err)
+		} else {
+			if err := os.Setenv("KUBE_MASTER_IP", strings.TrimSpace(string(masterIP))); err != nil {
+				return err
+			}
+		}
+
+		if os.Getenv("ENABLE_KUBEMARK_CLUSTER_AUTOSCALER") == "true" {
+			testArgs = append(testArgs, "--kubemark-external-kubeconfig="+os.Getenv("DEFAULT_KUBECONFIG"))
+		}
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		//util := os.Getenv("WORKSPACE") + "/kubernetes/cluster/kubemark/util.sh"
+		//testArgs = append([]string{"-c", "source", util, " ; ./hack/ginkgo-e2e.sh"}, testArgs...)
```

Comment:
Get rid of these commented lines (if they're not useful)?

Reply:
That's probably fine. But since this is a relatively risky change, let's keep an eye on the kubemark jobs while rolling this out and revert quickly if needed.